src/package-compatibility.js

```diff
@@ -127,19 +129,19 @@ export function checkOne(info: Manifest, config: Config, ignoreEngines: boolean)
   };
 
   const invalidPlatform =
-    !config.ignorePlatform && Array.isArray(info.os) && info.os.length > 0 && !isValidPlatform(info.os);
+    shouldCheckPlatform(info, config.ignorePlatform) && info.os != null && !isValidPlatform(info.os);
```

Comment:
Flow requires the `infos.os != null` check because it does not know that the `os` field is never null when `shouldCheckPlatform` is true. If we use TypeScript we could implement `shouldCheckPlatform` as a user defined type guard.

```
function isFish(pet: Fish | Bird): pet is Fish {
    return (<Fish>pet).swim !== undefined;
}
```

Reply:
Thanks @arcanis for referencing me to the documentation of `%checks` ([link](https://flow.org/en/docs/types/functions/#toc-predicate-functions)).